.tools/run_jvm_tests.sh

```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -eufx -o pipefail
```

Comment:
`set -e` is the crucial part - without it intermediate tests can be failing arbitrarily, and won't cause the overall workflow to fail!

Reply:
this also makes remembering to the explicitly `|| exit` all the tests redundant.